### DISCUSSION TITLE
network error when trying to DL subtitles

### DISCUSSION THREAD

**Discussion by derrellsimpson** [iina/iina#4664] [General]
hello.  I get a network error message when trying to DL subtitles form open subtitles.org.  I have tried using an account, I have tries setting up auto download, I have tried rebooting after making the changes.  still the same issue.  it finds three separate subtitles, but any one of them produces the network error message when I try to DL them.

have tried without VPN or ad-blocker with the same issue.

thanks in advance for any help!

macOS Sonoma, IINA 1.3.3 Build 138

DS

**Comment by low-batt** [comment]
Frequently this failure is due to exceeding the download quota when not using an account. As you mentioned you have an account, then it may be due to server overload.

A while back the logging in this area of IINA was improved, so possibly the log file will contain some information about what is going wrong.

To enable logging follow these instructions:
- Start IINA
- Click on  `Settings…` under the `IINA` menu
- The settings panel appears
- On the left side of the panel click on `Advanced`
- Slide the `Enable advanced settings` toggle button to be on (blue)
- Click on the checkbox `Enable logging to file` 
- Confirm the settings panel looks like the screenshot below
- **Restart IINA** to activate logging

![enable-logging](https://github.com/iina/iina/assets/86170219/f4d0b380-abca-4d9a-9eaf-56a1535127ff)

Once IINA has restarted, try downloading subtitles and if the failure reproduces then:

- Click on  `Settings…` under the `IINA` menu
- The settings panel appears
- On the left side of the panel click on `Advanced`
- Click on `Open log directory`
- A new `Finder` window appears containing the logs for this session
- **Quit IINA** to complete the log files

The `Finder` window contains two files, `iina.log` and `mpv.log`. To attach the log files to this issue, drag-n-drop them from the `Finder` window to the `Leave a comment` section below. Include both the `iina.log` and the `mpv.log`

The log files can also be accessed using [Terminal](https://support.apple.com/guide/terminal/welcome/mac) as shown here:
```bash
low-batt@gag com.colliderli.iina$ pwd
/Users/low-batt/Library/Logs/com.colliderli.iina
low-batt@gag com.colliderli.iina$ ls
2022-10-04-17-57-15_jVrbWy
low-batt@gag com.colliderli.iina$ cd 2022-10-04-17-57-15_jVrbWy
low-batt@gag 2022-10-04-17-57-15_jVrbWy$ ls
iina.log	mpv.log
low-batt@gag 2022-10-04-17-57-15_jVrbWy$ 
```

With logging enabled IINA creates a new log directory each time it is started. If accessing the logs using `Terminal` be sure to obtain them from the latest log directory.

**DO NOT FORGET** to disable logging once you have obtained log files for the problem being investigated. IINA does not restrict the size of log files or delete old logs. If you leave logging enabled the logs will continue to accumulate.

**Comment by derrellsimpson** [comment]
thanks!  i see the following errors in the log when trying to DL subtitles for this movie.  it finds three different subs, but all give the same network error:

08:16:43.073 [opensub][d] Searching for subtitles of movies matching 'The.Empty.Man.2020.1080p.AMZN.WEB-DL.DDP5.1.H.264-EVO.mkv'
08:16:47.575 [onlinesub][e] Failed to obtain subtitles for file:///Volumes/Seagate%20Backup%20Plus%20Drive/tv%20and%20movies/The.Empty.Man.2020.1080p.AMZN.WEB-DL.DDP5.1.H.264-EVO%5BTGx%5D/The.Empty.Man.2020.1080p.AMZN.WEB-DL.DDP5.1.H.264-EVO.mkv from opensubtitles.org. The operation couldn’t be completed. (IINA.OnlineSubtitle.CommonError error 1.)
08:19:54.013 [onlinesub][d] Search subtitle from opensubtitles.org...
08:19:54.212 [opensub][d] OpenSub: logged in as user derrellsimpson
08:19:54.213 [opensub][d] Searching for subtitles of movies matching hash ba71512e91182f07 and size 8755297231
08:19:54.388 [opensub][d] Searching for subtitles of movies matching 'The.Empty.Man.2020.1080p.AMZN.WEB-DL.DDP5.1.H.264-EVO.mkv'
08:19:58.302 [onlinesub][e] Failed to obtain subtitles for file:///Volumes/Seagate%20Backup%20Plus%20Drive/tv%20and%20movies/The.Empty.Man.2020.1080p.AMZN.WEB-DL.DDP5.1.H.264-EVO%5BTGx%5D/The.Empty.Man.2020.1080p.AMZN.WEB-DL.DDP5.1.H.264-EVO.mkv from opensubtitles.org. The operation couldn’t be completed. (IINA.OnlineSubtitle.CommonError error 1.)
08:20:00.714 [onlinesub][d] Search subtitle from opensubtitles.org...
08:20:00.902 [opensub][d] OpenSub: logged in as user derrellsimpson
08:20:00.902 [opensub][d] Searching for subtitles of movies matching hash ba71512e91182f07 and size 8755297231
08:20:01.066 [opensub][d] Searching for subtitles of movies matching 'The.Empty.Man.2020.1080p.AMZN.WEB-DL.DDP5.1.H.264-EVO.mkv'
08:20:05.892 [onlinesub][e] Failed to obtain subtitles for file:///Volumes/Seagate%20Backup%20Plus%20Drive/tv%20and%20movies/The.Empty.Man.2020.1080p.AMZN.WEB-DL.DDP5.1.H.264-EVO%5BTGx%5D/The.Empty.Man.2020.1080p.AMZN.WEB-DL.DDP5.1.H.264-EVO.mkv from opensubtitles.org. The operation couldn’t be completed. (IINA.OnlineSubtitle.CommonError error 1.)
08:20:12.835 [onlinesub][d] Search subtitle from opensubtitles.org...
08:20:13.012 [opensub][d] OpenSub: logged in as user derrellsimpson
08:20:13.013 [opensub][d] Searching for subtitles of movies matching hash ba71512e91182f07 and size 8755297231
08:20:13.185 [opensub][d] Searching for subtitles of movies matching 'The.Empty.Man.2020.1080p.AMZN.WEB-DL.DDP5.1.H.264-EVO.mkv'
08:20:15.835 [onlinesub][e] Failed to obtain subtitles for file:///Volumes/Seagate%20Backup%20Plus%20Drive/tv%20and%20movies/The.Empty.Man.2020.1080p.AMZN.WEB-DL.DDP5.1.H.264-EVO%5BTGx%5D/The.Empty.Man.2020.1080p.AMZN.WEB-DL.DDP5.1.H.264-EVO.mkv from opensubtitles.org. The operation couldn’t be completed. (IINA.OnlineSubtitle.CommonError error 1.)


![Screenshot 2023-10-22 at 8 23 31 AM](https://github.com/iina/iina/assets/74442157/fe3f0ac4-1ada-401b-82c4-323a2643d9ef)

**Comment by derrellsimpson** [comment]
I have tried with other MKVs as well with the same error.

any ideas?

thanks for the time and help!

DS

**Comment by low-batt** [comment]
The logs did not show what I expected. This is not the common failure.

I tired downloading myself, also using an account and experienced the same behavior. Search results returned fine, but download failed, just like for you.

I debugged IINA to see what is going on. Instead of the subtitles file Open Subtitles is returning a web page with the title `Just a moment...`. The web page says `Enable JavaScript and cookies to continue`.

I believe this is a [Cloudflare challenge](https://developers.cloudflare.com/firewall/cf-firewall-rules/cloudflare-challenges/). Cloudflare is trying to detect and refuse to serve robots. This kind of check **must not** be enabled on a URL that is part of an API used by applications. My working theory is that Open Subtitles accidentally enabled this breaking API access. Super busy today or I'd be checking the Open Subtitles forums to see if anybody has reported this to them. I will do that as soon as I can.

For the moment you will have to log into [opensubtitles.org](https://www.opensubtitles.org/) using a web browser and download the subtitles file directly and then get IINA to load it. I tried downloading from the web site and that was working.

**Comment by low-batt** [comment]
I posted the topic [Cloudflare Blocking XMLRPC Download](https://forum.opensubtitles.com/t/cloudflare-blocking-xmlrpc-download/2256) in the new [opensubtitles.com forum](https://forum.opensubtitles.com/).

**Comment by low-batt** [comment]
Open Subtitles has posted that they have corrected this problem. I tested downloading subtitles and it is back working for me.

**Comment by derrellsimpson** [comment]
Excellent!

DS
